RUFAS/EEE/emissions.py

```diff
@@ -283,387 +205,438 @@ def _get_feed_emissions_data(
+        for i, field_name in enumerate(filtered_data["field_name"]["values"]):
+            crop_name = filtered_data["crop"]["values"][i]
+            feed_id = crop_to_feed_id_mapping.get((field_name, crop_name), None)
+            harvest_dry_yield_data = filtered_data["dry_yield"]["values"][i]
+            harvest_type = filtered_data["harvest_type"]["values"][i]
+            harvest_year, harvest_day = filtered_data[year_key]["values"][i], filtered_data[day_key]["values"][i]
+            harvest_simulation_day = (
+                RufasTime.convert_year_jday_to_date(harvest_year, harvest_day) - simulation_start_date
+            ).days
+
+            result[field_name][harvest_simulation_day] = {
+                "field_name": field_name,
+                "crop": crop_name,
+                "feed_id": feed_id,
+                "dry_yield": harvest_dry_yield_data,
+                "harvest_type": harvest_type,
+            }
 
-        return sorted_crops
+        return result
 
-    def _partition_applied_crop_fertilizer_emissions(
-        self, fertilizer_application: dict[str, float], applied_crops: list[dict[str, Any]]
-    ) -> None:
-        """
-        Partitions synthetic emissions from fertilizer applications to crop(s) whose planting and harvesting dates
-        encompass the fertilizer application date.
-        """
-        for crop in applied_crops:
-            split_factor = 1 / len(applied_crops)
-            crop["nitrogen_fertilizer_used"] += fertilizer_application["nitrogen"] * split_factor
-            crop["nitrogen_fertilizer_embedded_CO2_emissions"] += (
-                fertilizer_application["nitrogen"] * split_factor * EMBEDDED_NITROGEN_FERTILIZER_EMISSIONS_FACTOR
-            )
-            crop["phosphorus_fertilizer_used"] += fertilizer_application["phosphorus"] * split_factor
-            crop["phosphorus_fertilizer_embedded_CO2_emissions"] += (
-                fertilizer_application["phosphorus"] * split_factor * EMBEDDED_PHOSPHORUS_FERTILIZER_EMISSIONS_FACTOR
+    def _calculate_daily_farmgrown_feed_emissions_and_resources(
+        self,
+        emission_data: dict[str, dict[str, dict[int, float]]],
+        resource_data: dict[str, dict[str, dict[int, dict[str, float]]]],
+        harvest_yield_by_field: dict[str, dict[int, dict[str, Any]]],
+        all_simulation_days: list[int],
+    ) -> dict[RUFAS_ID, dict[int, dict[str, float]]]:
+        """Calculates daily emissions and resources used for farmgrown feeds production."""
+
+        total_farmgrown_feed_emission_and_resource_by_feed_id: dict[RUFAS_ID, dict[str, float]] = defaultdict(dict)
+        total_harvest_dry_yield_by_feed_id: dict[RUFAS_ID, float] = defaultdict(float)
+        daily_farmgrown_feed_emission_and_resource_by_feed_id: dict[RUFAS_ID, dict[int, dict[str, float]]] = (
+            defaultdict(dict)
+        )
+
+        harvest_dates_by_feed_id: dict[RUFAS_ID, list[int]] = {
```

Comment:
these are all super helpful and descriptive to understanding all the parsing going on, I would model the others after these wherever possible.